crates/matrix-sdk-ui/src/timeline/tests/basic.rs

```diff
@@ -62,10 +62,12 @@ async fn test_initial_events() {
         )
         .await;
 
-    let item = assert_next_matches!(stream, VectorDiff::PushBack { value } => value);
-    assert_matches!(&item.kind, TimelineItemKind::Virtual(VirtualTimelineItem::DayDivider(_)));
     let item = assert_next_matches!(stream, VectorDiff::PushBack { value } => value);
     assert_eq!(item.as_event().unwrap().sender(), *ALICE);
+
+    let item = assert_next_matches!(stream, VectorDiff::Insert { index: 0 , value } => value);
+    assert_matches!(&item.kind, TimelineItemKind::Virtual(VirtualTimelineItem::DayDivider(_)));
```

Comment:
Is there a day divider because the ts moves from 0 to 1? (I'm looking at https://github.com/matrix-org/matrix-rust-sdk/blob/f42c8937daca8a091e33c3d98743d8c2eb5439f0/testing/matrix-sdk-test/src/event_builder.rs#L44 )

Either way, a comment to explain why these two events are on different dates would be helpful.

Also, I hope/assume this test passes if you run it in any time zone?

Reply:
Ah no, it's just that now the day divider is inserted *after* handling the events. We end up with the same final array shape, but instead of doing push_back (pb) day-divider / pb alice / pb bob, we now do pb alice / pb bob / push_front day-divider.